Content.Server/Atmos/Piping/Unary/EntitySystems/GasCanisterSystem.cs

```diff
@@ -73,6 +81,9 @@ private void OnCanisterStartup(EntityUid uid, GasCanisterComponent canister, Com
             {
                 containerManager.MakeContainer<ContainerSlot>(canister.ContainerName);
             }
+
+            if (TryComp<LockComponent>(uid, out var lockComponent))
+                _appearanceSystem.SetData(uid, GasCanisterVisuals.Locked, lockComponent.Locked);
```

Comment:
this should really be a generic thing that lockcomponent handles (and renamed to `LockVisuals` or something) but I won't enforce that it's done in this PR